packages/ppx_irmin/ppx_irmin.2.3.0/opam

```diff
@@ -0,0 +1,32 @@
+opam-version: "2.0"
+maintainer: "Craig Ferguson <craig@tarides.com>"
+homepage: "https://github.com/mirage/irmin"
+bug-reports: "https://github.com/mirage/irmin/issues"
+license: "ISC"
+dev-repo: "git+https://github.com/mirage/irmin.git"
+
+build: [
+  ["dune" "subst"] {pinned}
+  ["dune" "build" "-p" name "-j" jobs]
+  ["dune" "runtest" "-p" name "-j" jobs] {with-test}
+]
+
+depends: [
+  "dune" {>= "2.7.0"}
+  "ocaml" {>= "4.08.0"}
+  "ppxlib" {>= "0.12.0" & < "0.18.0"}
```

Comment:
I don't remember seeing the previous version of ppx_irmin incompatible with ppxlib >= 0.18.0. Is that expected? This is required for a proper compatibility with OCaml 4.12.

Reply:
Probably just because `ppxlib.0.18` wasn't out at the last Irmin release; this constraint was added to prior versions during the `0.18` Ppxlib release https://github.com/ocaml/opam-repository/pull/17360/files.

We'd require a cleverer fix to drop this upper bound, since we can't afford to lose compatibility with pre-0.18 versions of ppxlib at the moment.

Reply:
> We'd require a cleverer fix to drop this upper bound, since we can't afford to lose compatibility with pre-0.18 versions of ppxlib at the moment.

In the past this was done using `cppo`, however since Dune does not support pulling package version this is harder but not impossible (I've done it for the switch of dose3 to dune using ocamlfind + cppo)

Reply:
I'm happy to make a change like that, but it would require a re-release of `repr` since `ppx_repr` has the same issue. Perhaps we can save that for `2.3.1`?

Reply:
Sure, sure! ^_^